### PR TITLE
1758 make views stats on preview display more by default

### PR DESCRIPTION
**Pull request by ssethupathy**

### CODE REVIEW

File: sourcecode/www/src/components/ResourceModal/ResourceStats.jsx [hunk 1]
```diff
@@ -19,60 +19,88 @@ const ResourceStats = ({ resourceId }) => {
...
+            </Box>
+
+            {loading || !response ?
+                <CircularProgress/>
+                :
+                <ResponsiveContainer width="100%" height={100}>
+                    <LineChart
+                        data={fillEmptyDays(
+                            dateRangeViews,
+                            moment().subtract(diffDays, 'days').startOf('day'),
```
Comment (emmachughes) [inline sourcecode/www/src/components/ResourceModal/ResourceStats.jsx:70]: The second and third argument to `fillEmptyDays()` are not correct, as this code operates on the current day, but both the start and end date are now arbitrary.

File: sourcecode/www/src/components/ResourceModal/ResourceStats.jsx [hunk 2]
```diff
@@ -19,60 +19,88 @@ const ResourceStats = ({ resourceId }) => {
...
     const datasets = [
         {
             key: 'count',
             name: _.capitalize(t('resource_view', { count: 2 })),
-            dataset: last7daysViews,
+            dataset: dateRangeViews,
         },
     ];
 
+    const fromDate = moment(startDate);
```
Comment (emmachughes) [inline sourcecode/www/src/components/ResourceModal/ResourceStats.jsx:41]: `fromDate` and `toDate` are unnecessary and confusing, as they're copies of `startDate` and `endDate`, respectively, but are Moment objects instead of strings. Since Moment can operate on strings, these variables can be removed, and diffDays refactored as:

```js
const diffDays = moment(endDate).diff(fromDate, 'days')
```

File: sourcecode/apis/resources/src/controllers/stats.js [hunk 3]
```diff
@@ -12,10 +12,10 @@ export default {
     getResourceStats: async (req) => {
         return {
             data: {
-                last7daysViews:
+                dateRangeViews:
                     await req.context.db.trackingResourceVersion.getCountByDayForResource(
-                        moment().subtract(7, 'days').startOf('day').toDate(),
-                        moment().endOf('day').toDate(),
+                        moment(req.query.start).startOf("day").toDate(),
```
Comment (emmachughes) [inline sourcecode/apis/resources/src/controllers/stats.js:17]: Changing from single quotes to double quotes is unnecessary, and introduces inconsistencies with the rest of the file.

**Review by emmachughes — CHANGES_REQUESTED** [review]

File: sourcecode/www/src/i18n/nn/translation.json [hunk 4]
```diff
@@ -145,7 +145,9 @@
     "Sett inn": "Set inn",
     "S": {
         "ERROR_PUBLISHING": "Noko skjedde når me prøvde å publisera. Ver vennleg og prøv igjen eller kom tilbake seinare.",
-        "VIEWS_PAST_7_DAYS": "Visningar dei siste 7 dagar"
+        "VIEWS_PAST_7_DAYS": "Visningar dei siste 7 dagar",
+        "VIEWS": "Visninger",
```
[the remark this answers is not in the document]
Comment (ssethupathy) [reply sourcecode/www/src/i18n/nn/translation.json:149]: Comments addressed

File: sourcecode/www/src/i18n/en/translation.json [hunk 5]
```diff
@@ -80,7 +80,10 @@
     "Sett inn": "Insert",
     "S": {
         "ERROR_PUBLISHING": "Something went wrong while trying to publish. Please try again, or come back later.",
-        "VIEWS_PAST_7_DAYS": "Views last 7 days"
+        "VIEWS_PAST_7_DAYS": "Views last 7 days",
```
[the remark this answers is not in the document]
Comment (ssethupathy) [reply sourcecode/www/src/i18n/en/translation.json:83]: Review comments addressed

File: sourcecode/apis/resources/src/routes/stats.js [hunk 6]
```diff
@@ -1,8 +1,8 @@
 import express from 'express';
```
Comment (emmachughes) [inline sourcecode/apis/resources/src/routes/stats.js:1]: Please revert the changes in this file.

File: sourcecode/www/src/components/ResourceModal/ResourceStats.jsx [hunk 7]
```diff
@@ -8,71 +8,100 @@ import {
...
+
     return (
         <>
             <Box pb={2}>
-                <strong>{t('S.VIEWS_PAST_7_DAYS')}</strong>
+                <strong>{t('S.VIEWS')}</strong>
+            </Box>
+
+            <Box pb={2}>
+                From: <label htmlFor="from"> <input type="date" id="from" name="from" max={endDate}
```
Comment (emmachughes) [inline sourcecode/www/src/components/ResourceModal/ResourceStats.jsx:55]: The label text has to be *inside* `<label>`.

The `name` property isn't needed, since this isn't a real HTML form.

File: sourcecode/www/src/components/ResourceModal/ResourceStats.jsx [hunk 8]
```diff
@@ -8,71 +8,100 @@ import {
...
+                                                    onChange={handleDateChange}
+                                                    value={startDate}/> </label> {' '}
+                To: <label htmlFor="to"><input type="date" id="to" name="to"
```
Comment (emmachughes) [inline sourcecode/www/src/components/ResourceModal/ResourceStats.jsx:58]: Here, too.

File: sourcecode/www/src/components/ResourceModal/ResourceStats.jsx [hunk 9]
```diff
@@ -8,71 +8,100 @@ import {
     XAxis,
     YAxis,
 } from 'recharts';
-import { fillEmptyDays } from '../../helpers/chart.js';
```
Comment (emmachughes) [inline sourcecode/www/src/components/ResourceModal/ResourceStats.jsx:11]: Unrelated code style changes shouldn't be submitted. They make it harder to perform code reviews, and the original code ends up being misattributed (e.g. by `git blame`) to your commit.

PHPStorm can reformat portions of code. You should reformat only the stuff you added or changed.

**Review by emmachughes — CHANGES_REQUESTED** [review]

**Review by emmachughes — APPROVED** [review]
